File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        if not strs:
            return ""
        res = ""
        for s in strs:
            res += str(len(s)) + "#" + s
        return res

    def decode(self, s: str) -> List[str]:
        if not s:
            return []
        
        res = []
        i = 0

        while i < len(s):
            cnt = i
            while s[cnt] != "#":
                cnt += 1
            length = int(s[i:cnt])
            i = cnt + 1
            cnt = i + length
            res.append(s[i:cnt])
            i = cnt
        return res

            

File: Data Structures & Algorithms/pow-x-n/submission-0.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double res = 1;
        int cnt = abs(n);
        while(cnt){
            res *= x;
            cnt--;
        }
        if (n<0){
            return(1/res);
        }
        return res;
    }
};
